src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -578,19 +578,32 @@ def update_iot_hub_custom(instance,
     if fileupload_notification_ttl is not None:
         ttl = timedelta(hours=fileupload_notification_ttl)
         instance.properties.messaging_endpoints['fileNotifications'].ttl_as_iso8601 = ttl
-    # if setting a fileupload storage identity or changing fileupload to identity-based
-    if fileupload_storage_identity or fileupload_storage_authentication_type == AuthenticationType.IdentityBased.value:
-        _validate_fileupload_identity(instance, fileupload_storage_identity)
-
-    default_storage_endpoint = _process_fileupload_args(
-        instance.properties.storage_endpoints['$default'],
-        fileupload_storage_connectionstring,
-        fileupload_storage_container_name,
-        fileupload_sas_ttl,
-        fileupload_storage_authentication_type,
-        fileupload_storage_container_uri,
-        fileupload_storage_identity,
-    )
+    # only bother with $default storage endpoint checking if modifying fileupload params
+    if any([
+            fileupload_storage_connectionstring, fileupload_storage_container_name, fileupload_sas_ttl,
+            fileupload_storage_authentication_type, fileupload_storage_container_uri, fileupload_storage_identity]):
```

Comment:
what kind of combinations are possible here? EX: there is a connection string but no container name, etc

Reply:
As far as combinations - this check is just to see if any parameters the user is modifying require the `$default` endpoint to be created, as these are the updates that are failing if the endpoint doesn't exist.

In order to successfully recreate the endpoint in the same command, they need to add the connection string and the container name, and the endpoint can be recreated, otherwise it will throw an error that the endpoint does not exist, and must be recreated (with a sample command).